conductor_api/client.py

```diff
@@ -0,0 +1,398 @@
+
+            [{
+                 'created': '2023-05-17T18:10:23.637Z',
+                 'trackedSearchIds': [64109775, 94110693],
+                 'modified': '2023-05-17T18:10:23.637Z',
+                 'name': '2023 Targeted Keywords'
+             }]
+            """
+        return self._make_request(
+            f"{self._v3_url}/accounts/{self.account_id}/categories"
+        )
+
+    # Collection Data
+
+    def get_ranks(self, web_property_id, rank_source_id, date="CURRENT"):
+        """Retrieves the ranks for the specified web property and rank source
+        for a given date.
+
+        Args:
+            web_property_id (str): The ID of the web property to retrieve the search volume for.
+            rank_source_id (str): The ID of the rank source to retrieve the search volume for.
+            date (str): The date to retrieve the search volume for. Defaults to "CURRENT", which represents the current date.
+
+        Returns:
+            list[dict]: A list of the JSON responses containing the rank data
+
+        Example::
+
+         [{
+              'trackedSearchId': 53383270,
+              'webPropertyId': None,
+              'targetDomainName': 'www.amazon.com',
+              'itemType': 'STANDARD_LINK',
+              'target': 'Young Living Valor Essential Oil - Empowering Blend with a ...',
+              'targetUrl': 'https://www.amazon.com/example',
+              'ranks': {'TRUE_RANK': 3, 'UNIVERSAL_RANK': 2, 'CLASSIC_RANK': 2}
+          }]
+        """
+        time_period = week_number(date) if date != "CURRENT" else date
+        return self._make_request(
+            f"{self._v3_url}/{self.account_id}/web-properties/{web_property_id}/rank-sources/{rank_source_id}/"
+            f"tp/{time_period}/serp-items"
+        )
+
+    def get_volume(self, web_property_id, rank_source_id, date="CURRENT"):
+        """Retrieves search volume for the specified web property and rank source
+        for a given date.
+
+        Args:
+            web_property_id (str): The ID of the web property to retrieve the search volume for.
+            rank_source_id (str): The ID of the rank source to retrieve the search volume for.
+            date (str): The date to retrieve the search volume for. Defaults to "CURRENT", which represents the current date.
+
+        Returns:
+            list[dict]: A list of the JSON response containing the search volume data.
+
+        Example::
+
+         [{
+              'averageVolume': 2900,
+              'trackedSearchId': 53383249,
+              'volumeItems': [{'volume': 1000, 'month': 7, 'year': 2023},
+                              {'volume': 880, 'month': 6, 'year': 2023}]
+         }]
+         """
+        time_period = week_number(date) if date != "CURRENT" else date
+        return self._make_request(
+            f"{self._v3_url}/{self.account_id}/web-properties/{web_property_id}/rank-sources/{rank_source_id}/"
+            f"tp/{time_period}/search-volumes"
+        )
```

Comment:
Instead of adding it directly into the function, I would maybe think about storing this url in an url variable for better readability. You can also separate it better by having a base_url which is 

```suggestion
        time_period = week_number(date) if date != "CURRENT" else date
        endpoint = "/{self.account_id}/web-properties/{web_property_id}/rank-sources/{rank_source_id}/tp/{time_period}/search-volumes"
        url = self._v3_url + endpoint
        return self._make_request(url)
```

Something like this is way more readable for me at least.

Reply:
Also interesting that we do not have a 

"accounts/{account_id}/web-properties/{web_property_id}/..." 

but we are missing the accounts parameter. But maybe the url is designed like this.